src/crimson/os/seastore/lba_manager/btree/lba_btree_node_impl.cc

```diff
@@ -46,6 +46,8 @@ LBAInternalNode::lookup_ret LBAInternalNode::lookup(
     iter->get_val(),
     get_paddr()).safe_then([c, addr, depth](auto child) {
       return child->lookup(c, addr, depth);
+    }).safe_then([ref=LBANodeRef(this)](auto ret) {
```

Comment:
instead of adding yet another continuation for keep `this` alive, can we just capture it in the previous one? like
```c++
  return get_lba_btree_extent(
    c,
    meta.depth - 1,
    iter->get_val(),
    get_paddr()).safe_then([c, addr, depth, ref=LBANodeRef(this)](auto child) {
      return child->lookup(c, addr, depth);
    });
```

or capture it using a `finally()` call, so no need to consume and return `ret` explicitly, and it serves the purpose better, "finally" implies cleanup. like
```c++
  return get_lba_btree_extent(
    c,
    meta.depth - 1,
    iter->get_val(),
    get_paddr()).safe_then([c, addr, depth](auto child) {
      return child->lookup(c, addr, depth);
    }).finally([me=LBANodeRef(this)] {});
```

i prefer the former one as it saves the overhead of scheduling yet another task.

Reply:
Capturing it in the previous continuation is inadequate, the reference must outlive the future returned by lookup.  Switching it to a finally will work if you prefer.

Reply:
okay! could you switch to `finally` ?